src/core_plugins/elasticsearch/lib/clusters.js

```diff
@@ -0,0 +1,14 @@
+import Cluster from './cluster';
+
+const clusters = new Map();
```

Comment:
[@epixa](https://github.com/elastic/kibana/pull/9132/files#r92857935)
> Is this something we can store in the runtime state of the elasticsearch plugin rather than hardcoding it into the module context?  The current implementation, in practical terms, is really just a global singleton to the entire process.

[@tylersmalley](https://github.com/elastic/kibana/pull/9132/files#r93094885)
> We could instantiate Cluster with the servers object to persist the cache on the plugin, or instantiate with a map to use for caching but I don't see what that would resolve. Are we worried about how this would behave running on multiple threads?

[@epixa](https://github.com/elastic/kibana/pull/9132/files#r93112982)
> @tylersmalley Attaching the state to a runtime object like server means the logic for getCluster and createCluster can be used independently of the state.  The most immediate benefit of that is for testing, since the state becomes ephemeral and you can disregard it/start from scratch at will rather than worrying about cleaning things up in an afterEach (which, by the way, you wouldn't even be able to do properly in the current implementation since there's no way to remove clusters from the map).
> 
> As an alternative, this could just be a class or factory that we could invoke once during plugin init and use elsewhere.  To be honest, I'm not even sure how feasible this is in the current plugin model, though, so it might not even be in the cards.

[@spalger](https://github.com/elastic/kibana/pull/9132/files#r93119320)
> I'm more worried about the cluster objects not having the same lifecycle as the server. If the server is closed and reopened, what happens here? What happens if multiple servers run in the same thread and both register `'data'` clusters.